src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::AbstractVector{U}) where U
+    n = length(a)
+    bit = FenwickTree{U}(n)
+    @inbounds for i = 1:n
+        update!(bit, i, a[i])
+    end
+    bit
+end
+
+bit(F::FenwickTree{T}) where T = F.bit
+length(F::FenwickTree{T}) where T = F.n
+
+"""
+    update!(F::FenwickTree{T}, ind, val::T)
+
+Update the value of the [`FenwickTree`] by `val` from the index `ind` upto the length of the Fenwick Tree.
+
+""" 
+function update!(F::FenwickTree{T}, ind::Int, val::T) where T
```

Comment:
I would suggest renaming this to `inc!`, for consistency with accumulators [here](https://github.com/JuliaCollections/DataStructures.jl/blob/69c867fcccc8a8210b4d841ba0ea0b4e9487a660/src/accumulator.jl#L64-L70).

Reply:
If you do so, consider also adding a `dec!` function.